src/rust/ide/src/model/module/synchronized.rs

```diff
@@ -333,158 +358,234 @@ impl Deref for Module {
-            LsClientSetup { file_path,client,current_ls_code,current_ls_version}
+        /// Create a setup initialized with the data from the given mock description.
+        fn new_for_mock_data(data:&crate::test::mock::Unified) -> Self {
+            Self::new(&data.logger,data.module_path.clone(),data.get_code())
         }
 
-        fn expect_edit<EditApplier>(&self, result:json_rpc::Result<()>, edit_applier:EditApplier)
-        where EditApplier : FnOnce(&[TextEdit]) -> String + 'static {
-            let path       = self.file_path.clone();
-            let ls_version = self.current_ls_version.clone_ref();
-            let ls_code    = self.current_ls_code.clone_ref();
-            self.client.expect.apply_text_file_edit(move |edit| {
-                let new_content = edit_applier(&edit.edits);
-                // check the requested edit content:
-                assert_eq!(edit.path       , path);
-                assert_eq!(edit.old_version, ls_version.get());
-                assert_eq!(edit.new_version, Sha3_224::new(new_content.as_bytes()));
-                // Update state:
-                if result.is_ok() {
-                    ls_code.set(new_content);
-                    ls_version.set(edit.new_version.clone());
+        /// Set up general text edit expectation in the mock client.
+        ///
+        /// This edit can be anything (full invalidation, code or metadata edit). The given function
+        /// should perform any necessary assertions on the `FileEdit` and provide the result that
+        /// the mock client will reply with.
+        fn expect_some_edit
+        (&self, client:&mut MockClient, f:impl FnOnce(&FileEdit) -> json_rpc::Result<()> + 'static) {
+            let this = self.clone();
+            client.expect.apply_text_file_edit(move |edits| {
+                let content_so_far = this.current_ls_content.get();
+                let ret            = f(edits);
+                let new_content    = apply_edits(content_so_far, &edits);
+                let actual_old     = this.current_ls_version.get();
+                let actual_new     = Sha3_224::new(new_content.as_bytes());
+                debug!(this.logger,"Actual digest:   {actual_old} => {actual_new}");
+                debug!(this.logger,"Declared digest: {edits.old_version} => {edits.new_version}");
+                debug!(this.logger,"New content:\n===\n{new_content}\n===");
+                assert_eq!(&edits.path,this.path.file_path());
+                assert_eq!(edits.old_version,actual_old);
+                assert_eq!(edits.new_version,actual_new);
+                if ret.is_ok() {
+                    this.current_ls_content.set(new_content);
+                    this.current_ls_version.set(actual_new);
+                    debug!(this.logger,"Accepted!");
+                } else {
+                    debug!(this.logger,"Rejected!");
                 }
-                result
+                ret
```

Comment:
why this name? Earlier it was `result` and in other places you are still using `result`